packages/consumption/src/modules/requests/utility/validateRelationshipAttributesWithinRequest.ts

```diff
@@ -0,0 +1,162 @@
+    if (containsMustBeAcceptedDuplicatesResult.containsDuplicates) {
+        return ValidationResult.error(
+            ConsumptionCoreErrors.requests.violatedKeyUniquenessOfRelationshipAttributes(
+                `The Request cannot be created because its acceptance would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key '${containsMustBeAcceptedDuplicatesResult.duplicateFragment.key}', owner and value type.`
+            )
+        );
+    }
+
+    return ValidationResult.success();
+}
+
+export function validateKeyUniquenessOfRelationshipAttributesWithinIncomingRequest(
+    items: (RequestItem | RequestItemGroup)[],
+    params: (DecideRequestItemParametersJSON | DecideRequestItemGroupParametersJSON)[],
+    recipient: CoreAddress
+): ValidationResult {
+    const fragmentsOfMustBeAcceptedItemsOfRequest = extractRelationshipAttributeFragmentsFromMustBeAcceptedItems(items, recipient);
+
+    const containsMustBeAcceptedDuplicatesResult = containsDuplicateRelationshipAttributeFragments(fragmentsOfMustBeAcceptedItemsOfRequest);
+    if (containsMustBeAcceptedDuplicatesResult.containsDuplicates) {
+        throw ConsumptionCoreErrors.requests.violatedKeyUniquenessOfRelationshipAttributes(
+            `The Request can never be accepted because it would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key '${containsMustBeAcceptedDuplicatesResult.duplicateFragment.key}', owner and value type.`
+        );
+    }
+
+    const fragmentsOfAcceptedItemsOfRequest = extractRelationshipAttributeFragmentsFromAcceptedItems(items, params);
+
+    const containsAcceptedDuplicatesResult = containsDuplicateRelationshipAttributeFragments(fragmentsOfAcceptedItemsOfRequest);
+    if (containsAcceptedDuplicatesResult.containsDuplicates) {
+        return ValidationResult.error(
+            ConsumptionCoreErrors.requests.invalidAcceptParameters(
+                `The Request cannot be accepted with these parameters because it would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key '${containsAcceptedDuplicatesResult.duplicateFragment.key}', owner and value type.`
+            )
+        );
+    }
+
+    return ValidationResult.success();
+}
+
+function extractRelationshipAttributeFragmentsFromMustBeAcceptedItems(
+    items: (RequestItem | RequestItemGroup)[],
+    recipient?: CoreAddress
+): RelationshipAttributeFragment[] | undefined {
+    const fragmentsOfMustBeAcceptedItemsOfGroup: RelationshipAttributeFragment[] = [];
+
+    for (const item of items) {
+        if (item instanceof RequestItemGroup) {
+            const fragments = extractRelationshipAttributeFragmentsFromMustBeAcceptedItems(item.items, recipient);
+            if (fragments) fragmentsOfMustBeAcceptedItemsOfGroup.push(...fragments);
+        } else {
+            if (!item.mustBeAccepted) return;
```

Comment:
If the condition is true, a return would be made and the `for` loop would be ended so that the remaining `items` are no longer processed. The condition should therefore be added to the one below.